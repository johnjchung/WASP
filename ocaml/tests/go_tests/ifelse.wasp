Func int ifelse (int x) {
  int i = 1;
  if(x == 3) {i = 12;}
  	else {i = 13;}
  return i;
 }
Endpoint ifelsetest(GET:) {
 int a = ifelse(3);
}
