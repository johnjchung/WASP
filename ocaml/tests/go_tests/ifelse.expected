package main

import (
	"github.com/gin-gonic/gin"
	"math"
	"os"
)

func int_conv(anything interface{}) int {
	switch v := anything.(type) {
		case int:
			return v
		case float64, float32:
			var float_var = v.(float64)
			return int(math.Trunc(float_var))
		default:
			return 0
	}
	return 0
}

func float_conv(anything interface{}) float64 {
 	 switch v := anything.(type) {
		case int:
            return float64(v)
		case float64:
			return v
		default:
			return 0
	}
	return 0.0
}

type ifelsetest_struct struct {
	A int `db:"a" json:"a"`
}

func main() {
	router := gin.Default()
	router.GET("/ifelsetest", Get_ifelsetest)
	router.Run(":"+os.Getenv("PORT"))
}

func ifelse(x int) int {
	i := 1
	if x == 3 {
	i = 12
	} else {
	i = 13
	}
	return i
}

func ifelsetest_struct_wrapper() ifelsetest_struct {
	return ifelsetest_struct{A: ifelse(3)}
}

func Get_ifelsetest(c *gin.Context) {
	ifelsetest_struct := ifelsetest_struct_wrapper()
	c.JSON(200, ifelsetest_struct)
}
